### ISSUE TITLE
Immediate "Is still here?" quest after solving cycleway quest

### ISSUE THREAD

**Issue by cyclingcat** [westnordost/StreetComplete#2245]
Hi there,

I just installed the freshly released SC version 26.0 and am impressed about the speed of fetching the icons! **👍👍👍** Now many additional cycleway quests pop up (finally again some street-releated quests in my direct neighbourhood...) as stated in the release notes. I solved some of them, but in some cases the quest icons don't vanish, and if I click again on the icon, the "Is this still the cycleway situation here?" quest appears. This doesn't seem to apply to _all_ new appearing cycleway quests, but I observed this several times in a few minutes, so it shouldn't been too difficult to reproduce it. I will continue to watch this in the course of the next days, maybe there is a simple pattern behind this. (Let me know if you need further assistance.)

(By the way: Some footway and road surface quests now pop up which never appeared during the past months. I did a quick check with JOSM: It's absolutely correct that these quests appear because the corresponding tags are missing in the OSM database. But why haven't these quests popped up during the past months or even years?)

The cycling cat

**Comment by matkoniecz** [comment]
> Let me know if you need further assistance.

If you link specific affected object then it may be helpful (for example my area has no "Is this still the cycleway situation here?" quests).

**Comment by matkoniecz** [comment]
> But why haven't these quests popped up during the past months or even years?

Nearby footway detector got smarter - see https://github.com/westnordost/StreetComplete/pull/2231

**Comment by cyclingcat** [comment]
> Nearby footway detector got smarter - see #2231

Okay, must have missed this one. But road surface quests seem to have been improved as well: A small road nearby tagged with "surface=paved" now appears again. It never did (even after the introduction of the road surface refinement quest) and now it does 👍.

But this is not the main topic of this ticket, just an observation. I will watch the coming cycleway quests and try to find one which is not directly in front of my flat (for privacy reasons). It shouldn't be too difficult.

The cycling cat

**Comment by westnordost** [comment]
>the "Is this still the cycleway situation here?" quest appears. This doesn't seem to apply to all new appearing cycleway quests, but I observed this several times in a few minutes, so it shouldn't been too difficult to reproduce it.

Yes, this sounds like a bug, but what we need to know is exactly which answer you gave for the quest to pop up again.

**Comment by westnordost** [comment]
Ah, I just read about your privacy reasons. You can send the location to me via email, newton@westnordost.de. I should still have PGP set up correctly in my email client. This is my public key:

-----BEGIN PGP PUBLIC KEY BLOCK-----

xsBNBFLgw1gBCADWynXA/ZMMBv0HwOALwcTGwfTAaVhEmgqrZ02mI7qoV8EBGG7aay0OcyqZOzeK
d+ADeYkukccijcBUWD0Sc1GaUThexGE0rdyUXkZUoW/AskBvCVnIT2MJOqvcMw0NtGu8x+7OTevH
4dMtIcDBXT7BSkbeHkPAUpDfFsy5QrVrNnXu5J+/IOrDlfvC76OyEWuGZLjbgvCHFsj+L1wibegs
o0QWMK89+ZGBYwJwL7jQBxEVPSP+rVR/R4dU+S6iGLkm9AzhSfsgj5j6Jqdk15CNrx96nWN8DPo7
RgEwqYSv/ZUxXOOhK05W3zDlkMyU+T3L1RyZak9RRBpE9EKXXhvXABEBAAHNI1RvYmlhcyBad2lj
ayA8bmV3dG9uQG9wZW5jbG9uay5vcmc+wsB+BBMBAgAoAhsjBgsJCAcDAgYVCAIJCgsEFgIDAQIe
AQIXgAUCXEMspwUJEOc3TwAKCRBFRw0mQ/ticjSkB/sGBs3CygzDW8ZIQDLFhEgvj+7rGZ/z+Jhe
GqTIBb9hEf94c9mktK9T7bp7nt2eqC046IY6QgQ7RX46/D7QJN0Y9hSClTE4iSUbGECSxTtB6+2U
UbXat/3p3W6mNWdp1FNzB9fSOpBNfiU8FO2E85bX6Wj8NY+SexmNIgRA2ecn+q0WMopVF50gt7xs
YWF/YGyGH7MmIijnv7NGQuHti++M0tJQahhahXUHFwj1L8N5OUf1fx5LRQfM0YCOUE7Qny6b2/hc
qvHJutZAKOcmZO5i5LdDanv29rQPJVxJGsz74dQNd9ltrlLX5Hlemrg6D46GdEQDekTBGDLlal8a
Kiy4wsB+BBMBAgAoBQJS4MQcAhsjBQkJZgGABgsJCAcDAgYVCAIJCgsEFgIDAQIeAQIXgAAKCRBF
Rw0mQ/ticvuJB/9lOanClK/t5TdMOMwVfvtcMnefRemsIONbw21ux7weUIkEV7fpgCV19oC+Wqly
goXt0YDkdmnVj7++v5RHgA4gI2kaW2bhZsz+g5I4r2pFn/s/zAcE99KmN7bzTIZ0a6QHbkGJV7B5
pOB/xYuWVH2m5Y2Nde+QoY+HP6UhMOOCsGGQhVA16ov5ZdHDkFsSK04LgRxCa2caCaIVucKExs7f
/wJicqzSfAAk5HPjjZkOu/vBiT2KuIWd2JQUjm3SZuRD+FI3kvjqzRM70+khqgv1wP1j50Q7DEKG
6T900/HmAk1P5JY5x9ZAoeMQQj+gDCl40FzOWrecR3qn3rFLykx3zSRUb2JpYXMgWndpY2sgPG5l
d3RvbkB3ZXN0bm9yZG9zdC5kZT7CwIEEEwECACsCGyMGCwkIBwMCBhUIAgkKCwQWAgMBAh4BAheA
AhkBBQJcQyynBQkQ5zdPAAoJEEVHDSZD+2JyRSoIALELRf1NEojzyG10t6tdtOIjORRM5Waiaf/O
ClrjCLX63BtImpIYnsQxyGnI2gnkY8HKfMFmiZxZLFkFmzlV/pBwGjVbGAwiCPjs+vcch5h02koJ
5U0nhWs8R7aDwLlTzMC1UE6nVhaCZw9D/1z17kQIKNYOyKagjUVtfmWd8G3lidk2AGBg4HwSY0tC
OOJEkP9DuorQm2kj7yPJHJPA/Xi5+EBCWIoWpnZ6xIgK+3T0Im7PnlljCEfup+n0tXXIe5sDtfiV
pzadspv39ifhxsEz8kvxsgFNIAMcQVVNkz2gAZt2TOoHpw5t220nDCBHzwS9Fwb9P7girc3wpuaR
pUfCwIEEEwECACsCGyMFCQlmAYAGCwkIBwMCBhUIAgkKCwQWAgMBAh4BAheABQJS4MQsAhkBAAoJ
EEVHDSZD+2Jy91sH/iDyC/0sOsiomlA3byJyp1y7Yxle6T9j7QeW7kC+jH3+NN7fANAllpnU5SFS
jooVa+SsuNLgsVvPFT2RIrCJTn1t4mqXlrK+usxIrBYU59q0R19uU99lX2GbHiR95fBgodLDDCfA
EWX6EccEBZ04QXGmbYVfu9e7Y2ya4od/7b0hszsFxbaRZD1Hi2ux2HlWcebB4vvv7RoY0SrG7Sno
OR4dOQ0YPgcOuR8Oau4aiQIYCKj2l6B9japXXdowu+PJAYE+h+B+zJ52sc7+mzIojxe7df9KG/cD
L++CQoCDt504T647f6DMScMxnbdSBYvN8rxADNn2CXFwtPcAhDUStrjCwH4EEwECACgFAlLgw1gC
GyMFCQlmAYAGCwkIBwMCBhUIAgkKCwQWAgMBAh4BAheAAAoJEEVHDSZD+2JyqEgH/jjI9BmbQALO
+2O2TZ4ruNmKJuxQC0MFPSJo+8gCM1LaMWGgJqRkbVoTgOjRjMJ2b99qYCfAb4LsizW+/pkOpXiL
RW14SAxslAjcUEQTwQPqRhxDStaae9lWn0T9O+maLF0gEDfut2RuQRP/z+/pM19Yk0lCTTgov+qP
6bhSy5wURyk9SqENgiuBZFgsAyYif5EOVPZ3BtzzY53/euByALO0bjTnX21hrh+6gM39SjOlvlk9
n5tRltepzlBCnAi9Dc2S1Q0JS7kB3V5hRjGOkW68z+FS1YsIh64AlCF9LYRA096Ed1peQh3q9mDY
SFIptTDgveSCidYDoBH3NIk08RnNIVRvYmlhcyBad2ljayA8b3NtQHdlc3Rub3Jkb3N0LmRlPsLA
fwQTAQIAKQIbIwcLCQgHAwIBBhUIAgkKCwQWAgMBAh4BAheABQJcQyynBQkQ5zdPAAoJEEVHDSZD
+2JycAwH/idEcaikPwu8daog7Rp/wxp8DNohpdtlyDJzYWklIT0F4drT+ZksHqHAU4QcPcXpx7kV
4nG/E/8opgpY0XMStm+NkrKsG2+JjnK5p8QjmeHmsj4s8ymvoB+5CnMBKXw5ZhWok/TcZXFHhIft
YnhfjxBvjpp+hY2ijzobk6chlr80/6BsFBbq6y9/mvlK/X/mlJkvimR0agtMTCL32PmIxcvI+yLt
Z3G8W61CJS6vDfrLHiAh+AIvs/KJ/UC6DC1uoOKiQl6e1awU30VI0OKbCFCT4+WZXmCSDJxi/Fyy
wJdepJRU93vAC6lk5c+RR8KGTEQQFLeW1szE0CSPh3KDtcvNIlRvYmlhcyBad2ljayA8YWRtaW5A
b3BlbmNsb25rLm9yZz7CwH8EEwECACkCGyMHCwkIBwMCAQYVCAIJCgsEFgIDAQIeAQIXgAUCXEMs
pwUJEOc3TwAKCRBFRw0mQ/tickzEB/9ASyip8qvWJtJ4j0ErAiFKVyT5/Hx7a3TIGp3GW5snC/sH
eerMcDLSyoC5mNja5rUu3yJtK+g052Qx3W7iLy7XWDP6sBuTv47QCDLBAD2Isk2uG7undx9u92Wn
LY9EWAumhyH5XJ47ymm2ZmhVK9/RiHswJY0tX/Dh0hgJTGRD0Vee1ZPEdETw4+4mAwWo9oao203z
Cg8aGohT2txU5crciX1nJm3vy9WoajJDLGmnZwu5mYzQN5Bu7ele1ydhU2hBpDt7VBG9sYAhcoHq
VQ4PoMoK4WQ+gCyQXkv2QOY3KI0m0vrI+rpxMgJ+KYWLtYXcsN8xJtenAlx8K4WPTqO+zsBNBFLg
w1gBCADWbzlpqg9SzVZ1B2vq0Yj6E6g/kT08im21Z+UlAh2FWS16kBNyXHd1pfzIE1Vi6NK7gDGK
38r/slOWFZZ0kOaFJYP+q+GWo9uRcSDJmnhzZHsFeBjmAsxI96KdT/FebZmDSyUZpHxFtMovRJyH
Vl2OJkkL1NU2UDiCSigNhsQFt5BM3M4f3FpBZqXSdXgKh7S56HTZZd3WzVvVpR4RtLSstGFvpDEe
BD+4/XAnmA0MEAbK7y3KbgkxcZiiD1rMLPSdm1mXhZZQgQQ0TuGG85pfcMa5ehwmtepi3x8uL9Ra
4StabdI2RHX6GV2B4L+ZbYCokFUkWfvaDvSUzvBmVEK5ABEBAAHCwGUEGAECAA8CGwwFAlxDLKcF
CRDnN08ACgkQRUcNJkP7YnKddAgAwgYR1Und2qYAbaYf/MgyFtWtyxTmV0HkapTM98/5dNrDShmk
3zon9vdHtYop/96HpnabMmJg4zLHinxaZKHsTtJ9Rxzrg+xx52rBiGGyS9/VtnetCa/1lqLlkGJY
A7ObVYloFluEvTz9VDL0SlQtDRW0OQOsqEJ6jMlRhl+a4T5kEUySwJ4S2Z+tbpQGCfTlh8MPr1aY
BGsUQO4qjznhGRpU6uOWGSBMTynj6voEGeuTTKX4jFkuoYCBrSNKe9Zs2bBEA3+n1p/qN4p4XuSM
kjqCNvtkTiD3HpzTL/7/x+osI5uDyBPkLU5zEYoLgY5bKSevWfI/LzCD4X56rbAIZMLAZQQYAQIA
DwUCUuDDWAIbDAUJCWYBgAAKCRBFRw0mQ/ticnXiCACcIznzuP3NTN6kZrsARIHXilthJHimMrg9
L15G33g/RGYYB+MjH43xYgeKrLRWDJTzcCggoyv6SU6y/38ibtu0MCdH9VK6fKqFk2WtJO+RGI1h
cIs0LOPZjcvMIsK5KD7WXBAeUwZ2ga+wv3viOjbaRyFFbjR8XSjPVInIB+9jeYGMlaG4UADdn3UJ
8Wcl/tr96g34XxklVmOdCnu6dvmFBpoo75uOeYIU6WEzQkN57Hgmjvy24uS52XU2mXlOy7wRu9d5
uUHF9r4tuxq2X5q6QPgfL7GQwOHyb6DcS4aKQoBFKsk4PoAAGY8dQwaG2IRdV62ibmk0TWGjA2iw
rqjf
=fe3J
-----END PGP PUBLIC KEY BLOCK-----

**Comment by cyclingcat** [comment]
Today I solved about 100 cycleway quests and didn't succeed in reproducing this issue. Sh*t! Don't know what happened this morning (but it wasn't a single occurance!). So I will continue keeping an eye on it during the next days and post it here if it will happen again.

(I don't use PGP though, just don't want to avoid that too much private information is disclosed publicly, e.g. via search engines. If I fail to reproduce the issue the next days I will mail the ID of the corresponding changeset to @westnordost.)

The cycling cat

**Comment by westnordost** [comment]
Do you remember which answer you gave that reproduced this bug? Certainly it was not `cycleway:both=n`o because you can't reproduce it with that.

**Comment by cyclingcat** [comment]
It was definitely `cycleway:both=no` as there is nothing else in the immediate vicinity of my home.

The cycling cat

**Comment by westnordost** [comment]
and you have auto-sync on?

**Comment by cyclingcat** [comment]
Yes, I have.

The cycling cat.

**Comment by westnordost** [comment]
Well, no idea then. The only thing I think could have happened is that your answer couldn't be applied due to a conflict (someone else edited the element in a way so that the change could not be applied). This would make the quest pin appear again.
I am going to close this, feel free to reopen if you find you can reproduce it.

**Comment by smichel17** [comment]
FYI, when you have closed the issue, only collaborators on the repo are able to reopen it, not the original author. Of course, the sentiment still applies, it's just , "comment here and I will reopen the issue" instead.

**Comment by westnordost** [comment]
Right, okay, didn't know that, thanks.

**Comment by cyclingcat** [comment]
Hi again,

just for the records: Some 100 solved quests later I still haven't succeeded in reproducing this issue. However, I noticed two probably related display quirks appearing every now and then, and if you combine them you'll get exactly ths issue in the ticket.

1. When I solve a road surface quest, sometimes a tracktype quest icon appears immediately, just to vanish again one second later. In rare cases the tracktype quest remains visible for a longer time (5 seconds or so). I have observed this behaviour with some other quest types as well (e. g. stairs), the second quest is always one related to the first one and it's probably one that is _not_ to be solved as it never appears again.

2. Occasionally I solve a quest, but the quest icon remains nevertheless visible. As soon as I pan the map, the icon disappears immediately.

So far I haven't found a pattern behind both mentioned effects, In any case it must be some rare conditions. But if you combine both effects you'll get a second quest that remains forever :smile: (I didn't check last week what would have happened if I had just panned the map though.)

The cycling cat
